File: 08-DesignPatterns-Java/Starter/SimpleFactory/src/Game.java
public final class Game
{
    private static int PipsOnDice = 6;

    public void play()
    {
        var dice = new Dice(PipsOnDice);

        var roll1 = dice.roll();
        var roll2 = dice.roll();
        var sum = roll1 + roll2;
        var hasWon = sum == 2 * PipsOnDice;
        var message = hasWon ? "You win!" : "You loose!";

        print(message);
    }

    private static void print(String message) {
        System.out.println(message);
    }
}

File: 10-DesignPatterns-CSharp/Solution/SimpleFactory/Game.cs
namespace SimpleFactory
{
    using System;

    using SimpleFactory.Factories;

    public sealed class Game
    {
        private const int PipsOnDice = 6;
        private readonly IDiceFactory _diceFactory;

        public Game(IDiceFactory diceFactory)
        {
            _diceFactory = diceFactory;
        }

        public void Play()
        {
            var dice = _diceFactory.CreateDice(PipsOnDice);

            var roll1 = dice.Roll();
            var roll2 = dice.Roll();
            var sum = roll1 + roll2;
            var hasWon = sum == 2 * PipsOnDice;
            var message = hasWon ? "You win!" : "You loose!";

            Print(message);
        }

        private static void Print(string message) => Console.WriteLine(message);
    }
}
